owncloudApp/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java

```diff
@@ -207,6 +228,30 @@ public void sendDownloadedFile(OCFile ocFile) {
         }
     }
 
+    public void sendDownloadedFile(List<OCFile> ocFiles) {
+        if (!ocFiles.isEmpty()) {
+            Intent sendIntent = makeActionSendIntent(ocFiles);
+            // Show dialog, without the own app
+            String[] packagesToExclude = new String[]{mFileActivity.getPackageName()};
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                Intent shareSheetIntent = new ShareSheetHelper().getShareSheetIntent(
+                        sendIntent,
+                        mFileActivity.getApplicationContext(),
+                        R.string.activity_chooser_send_file_title,
+                        packagesToExclude
+                );
+
+                mFileActivity.startActivity(shareSheetIntent);
+            } else {
+                DialogFragment chooserDialog = ShareLinkToDialog.newInstance(sendIntent, packagesToExclude);
+                chooserDialog.show(mFileActivity.getSupportFragmentManager(), FTAG_CHOOSER_DIALOG);
```

Comment:
Does the old way work with several files?

Reply:
Yes, I think there is nothing to change there. Only the Intent